docs/Toy/examples/scalar.toy

```diff
@@ -6,8 +6,8 @@ def main() {
 }
 
 # CHECK:         "toy.func"() ({
-# CHECK-NEXT:      %0 = "toy.constant"() {"value" = dense<5.500000e+00> : tensor<f64>} : () -> tensor<f64>
-# CHECK-NEXT:      %1 = "toy.reshape"(%0) : (tensor<f64>) -> tensor<2x2xf64>
```

Comment:
this is sus, we can't reshape from 1xf64 to 2x2xf64, I must have messed up somewhere, I'll try to fix this quickly

Reply:
I had a look, and this change is explicitly incorrect, at least according to MLIR:

```sh
$ toyc-ch2 docs/Toy/examples/scalar.toy -emit=mlir
> module {
>   toy.func @main() {
>     %0 = toy.constant dense<5.500000e+00> : tensor<f64>
>     %1 = toy.reshape(%0 : tensor<f64>) to tensor<2x2xf64>
>     toy.print %1 : tensor<2x2xf64>
>     toy.return
>   }
> }
```

Reply:
https://mlir.llvm.org/docs/Dialects/Builtin/#example-6

Reply:
Do you know which of the changes below made this change as consequence? We should add an explicit unit test for the builder to preserve this behaviour, it's a bad sign that Toy tests caught it

Reply:
I added explicit support for zero-rank tensors, so this change is reverted!

Reply:
I will also add some more extensive filetests tomorrow!